metaseq_cli/train.py

```diff
@@ -402,7 +402,11 @@ def validate_and_save(
         )
 
     do_save = (
-        (end_of_epoch and epoch_itr.epoch % cfg.checkpoint.save_interval == 0)
+        (
+            end_of_epoch
+            and epoch_itr.epoch % cfg.checkpoint.save_interval == 0
+            and not cfg.checkpoint.no_epoch_checkpoints
```

Comment:
Hm... this flag is still a bit confusing - likely since there's just way too many no_* checkpointing related flags (https://github.com/facebookresearch/metaseq/blob/414c32dfbc0789421aeceb6d979fb0f214e82dc1/metaseq/dataclass/configs.py#L514-L530). Any chance we can clean all of these up?

Reply:
You can clean it up if you want but maybe the thing to do is change the help msg for no_epoch_checkpoints is "don't store checkpoints at epoch boundaries"

Reply:
Yeah let's start with that help message change for now for this PR. I'll open an issue to track the above cleanup.

Reply:
Give me the conditional approve then?